pilot/pkg/networking/core/listener_inbound.go

```diff
@@ -189,6 +189,20 @@ func (lb *ListenerBuilder) buildInboundHBONEListeners() []*listener.Listener {
 	return []*listener.Listener{terminate, l}
 }
 
+func (lb *ListenerBuilder) sanitizeFilterChainForHBONE(c *listener.FilterChain) {
+	fcm := c.GetFilterChainMatch()
+	if fcm != nil {
+		// Clear out settings that do not matter anymore
+		fcm.TransportProtocol = ""
+	}
+	if fcm == nil {
+		fcm = &listener.FilterChainMatch{}
+		c.FilterChainMatch = fcm
+	}
+	// Filter to only allowed ranges. This ensures we do not get HBONE requests to garbage IPs
+	fcm.PrefixRanges = []*core.CidrRange{util.ConvertAddressToCidr(lb.node.IPAddresses[0])}
```

Comment:
nit: should this map the slice of IPAddresses?

Reply:
Let me look into how we do it in other places

Reply:
:+1: looks safe. Thanks!